src/lib.rs

```diff
@@ -741,6 +788,25 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_update_timestamps() {
+        let td = tempfile::tempdir().unwrap();
+        let d = openat::Dir::open(td.path()).unwrap();
+        d.ensure_dir("foo", 0o755).unwrap();
+        let before = d.metadata("foo").unwrap();
+        while nix::unistd::sleep(2) == 2 {}
```

Comment:
Minor: since we're comparing nanoseconds anyway, let's avoid sleeping at all?

Reply:
I had to apply a disproportionate amount of `sync` to this test logic in order to let it work on GH runner (I guess because of stacked filesystems). In the end I think we'd really want to perform a `fsync(dirfd)` here, but unfortunately it is currently missing from libc-rs.

Reply:
https://github.com/coreos/openat-ext/pull/31 should make this test less ugly. I'll rebase this when that merges.